client/client.go

```diff
@@ -1397,7 +1397,11 @@ func (c *Client) collectHostStats() {
 			c.resourceUsageLock.Lock()
 			c.resourceUsage = ru
 			c.resourceUsageLock.Unlock()
-			c.emitStats(ru)
+
+			// Publish Node metrics if operator has opted in
+			if c.config.PublishNodeMetrics {
```

Comment:
It looks like you check the config settings inside the function. Are you sure you need the guard up here?